cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/runtime/batch/BasicMapReduceContext.java

```diff
@@ -77,6 +79,7 @@ public BasicMapReduceContext(Program program,
                                MapReduceSpecification spec,
                                long logicalStartTime,
                                String workflowBatch,
+                               @Nullable String adapterName,
```

Comment:
Probably better to just add new parameter to the end rather than sneak it in the middle to avoid merge conflict and update to existing code.